16/danielmichaels/weather.py

```diff
@@ -0,0 +1,72 @@
+import datetime
+import sys
+
+from api_secrets import WEATHER_TOKEN
+
+import requests
+import requests_cache
+
+from requests.exceptions import RequestException
+
+
+class Weather:
+    """A simple class that pulls in weather data from the openweathermap api and formats the response.
+
+    """
+
+    def __init__(self, city="london", country="gb"):
+        self.city = city
+        self.country = country
+        self.url = (
+            "http://api.openweathermap.org/data/2.5/weather?q={city},"
+            "{country}&units=metric&appid={TOKEN}".format(
+                city=self.city, country=self.country, TOKEN=WEATHER_TOKEN
+            )
+        )
+        self.get_data
+
+    @property
+    def get_data(self):
+        """Get the weather data in JSON format. Caches the response for
+        600 seconds in a sqlite database.
+
+        """
+        requests_cache.install_cache(expire_after=600)
```

Comment:
love `requests_cache`, I would add this at the top or at least put this "magic number" in a constant

Reply:
Agreed, this doesn't make much sense here.